router/pkg/connectrpc/handler.go

```diff
@@ -0,0 +1,629 @@
+	operation := h.operationRegistry.GetOperationForService(serviceName, methodName)
+	if operation == nil {
+		// Log all available operations for this service to help diagnose the issue
+		allOps := h.operationRegistry.GetAllOperationsForService(serviceName)
+		var availableOps []string
+		for _, op := range allOps {
+			availableOps = append(availableOps, op.Name)
+		}
+		h.logger.Error("operation not found",
+			zap.String("service", serviceName),
+			zap.String("requested_method", methodName),
+			zap.Strings("available_operations", availableOps))
+		return nil, fmt.Errorf("operation not found for service %s: %s", serviceName, methodName)
+	}
+
+	h.logger.Debug("resolved operation",
+		zap.String("service", serviceName),
+		zap.String("rpc_method", methodName),
+		zap.String("operation", operation.Name),
+		zap.String("type", operation.OperationType))
+
+	// Convert proto JSON (snake_case) to GraphQL variables (camelCase)
+	// Pass service and method info for schema-aware enum detection
+	variables, err := h.convertProtoJSONToGraphQLVariables(serviceName, methodName, requestJSON)
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert proto JSON to GraphQL variables: %w", err)
+	}
+
+	// Execute the GraphQL query
+	responseJSON, err := h.executeGraphQL(ctx, operation.OperationString, variables)
+	if err != nil {
+		return nil, fmt.Errorf("failed to execute GraphQL query: %w", err)
+	}
+
+	return responseJSON, nil
+}
+
+// convertProtoJSONToGraphQLVariables converts proto JSON to GraphQL variables:
+// - Converts field names from snake_case to camelCase
+// - Strips proto enum prefixes (MOOD_HAPPY -> HAPPY) using schema-aware detection
+func (h *RPCHandler) convertProtoJSONToGraphQLVariables(serviceName, methodName string, protoJSON []byte) (json.RawMessage, error) {
+	if len(protoJSON) == 0 {
+		return json.RawMessage("{}"), nil
+	}
+
+	var protoData map[string]any
+	if err := json.Unmarshal(protoJSON, &protoData); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal proto JSON: %w", err)
+	}
+
+	// Get proto message descriptor for schema-aware enum detection
+	var messageDesc protoreflect.MessageDescriptor
+	if h.validator != nil && h.validator.protoLoader != nil {
+		method, err := h.validator.protoLoader.GetMethod(serviceName, methodName)
+		if err == nil && method.InputMessageDescriptor != nil {
+			messageDesc = method.InputMessageDescriptor
+		}
+	}
+
+	graphqlData := h.convertKeysRecursive(protoData, messageDesc)
+
+	graphqlJSON, err := json.Marshal(graphqlData)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal GraphQL variables: %w", err)
+	}
+
+	return graphqlJSON, nil
+}
+
+// convertKeysRecursive converts map keys from snake_case to camelCase
+// and strips proto enum prefixes using schema information when available
+// Omits fields with empty string values (e.g., from _UNSPECIFIED enum values)
+func (h *RPCHandler) convertKeysRecursive(data any, messageDesc protoreflect.MessageDescriptor) any {
```

Comment:
I don't think this is right. We cannot convert proto and graphql back and forth without introspecting each schema and doing a lookup on how to name a key.